receiver/dotnetdiagnosticsreceiver/config.go

```diff
@@ -0,0 +1,45 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package dotnetdiagnosticsreceiver
+
+import (
+	"time"
+
+	"go.opentelemetry.io/collector/config/configmodels"
+)
+
+type Config struct {
+	configmodels.ReceiverSettings `mapstructure:",squash"`
+	// Optional directory name to save binary blobs read from the socket. This can
+	// be useful for troubleshooting/debugging. If left blank or unset, no blobs are
+	// saved.
+	BlobDir string `mapstructure:"blob_dir"`
+	// The maximum number of blob files to keep in the blob_dir, if specified. Defaults
+	// to 10 (most recent).
+	MaxBlobFiles int `mapstructure:"max_blob_files"`
+	// The process ID of the dotnet process from which to collect diagnostics. This
+	// receiver is intended to be used with an observer and receiver creator for
+	// process discovery.
+	PID int `mapstructure:"pid"`
```

Comment:
Do you connect using PID or ip:port?

Reply:
We connect using a Unix domain socket (on Mac/Linux, named pipe on Windows), which for this receiver is a file that lives in a temp directory. We find the file by using the PID and a naming convention. You can see the implementation here: https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/2200/files#diff-daa2805f68eb64a575a82c688fd19dcb2a649a1409f6d014ec0e9b884f768873R28

Reply:
Please document how this is used to discover the file you mentioned.

Reply:
Sure, I've added an explanation about the socket file.